users/serializers.py

```diff
@@ -1,61 +1,105 @@
 from rest_framework import serializers
-from .models import CustomUser, UserProfile 
+from django.contrib.auth import authenticate
+from users.models import User, Profile
+from users.enums import UserRole
+from users.validators import is_email, validate_unique_email, validate_unique_phone, validate_required_email_and_phone
 
-
-#Serializer for the CustomUser model
+class ProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Profile
+        fields = ["bio", "avatar"]
+        
 class UserSerializer(serializers.ModelSerializer):
+    profile = ProfileSerializer(required=False)
+
     class Meta:
-        model = CustomUser
-        fields = ['id', 'username', 'email', 'password', 'first_name', 
-             'last_name', 'phone_number', 'role']
-        extra_kwargs = {"password": {"write_only": True}}
+        model = User
+        fields = [
+            "id", "first_name", "last_name", "email", "phone", "role",
+            "created_at", "profile"
+        ]
+        read_only_fields = ["id", "role", "created_at"]
+        
+class RegisterSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(write_only=True, min_length=8)
+    email = serializers.EmailField(validators=[validate_unique_email])
+    phone = serializers.CharField(validators=[validate_unique_phone])
 
+    class Meta:
+        model = User
+        fields = ["first_name", "last_name", "email", "phone", "password"]
+        
+    def validate(self, attrs):
+        return validate_required_email_and_phone(attrs)
+    
     def create(self, validated_data):
-        user = CustomUser.objects.create_user(**validated_data)
+        password = validated_data.pop("password")
+        user = User.objects.create_user(password=password, **validated_data)
+        Profile.objects.create(user=user)
```

Comment:
Wrap these database operations in a transaction to ensure atomicity, which helps roll back changes if any of the operation fails.